rct229/rulesets/ashrae9012019/ruleset_functions/get_energy_required_to_heat_swh_use.py

```diff
@@ -0,0 +1,221 @@
+                "design_supply_temperature",
+            )
+            inlet_temperature_schedule_id = getattr_(
+                distribution_system,
+                "service_water_heating_distribution_systems",
+                "entering_water_mains_temperature_schedule",
+            )
+            inlet_temperature_schedule = find_exactly_one_schedule(
+                rmd, inlet_temperature_schedule_id
+            )
+            inlet_temperature_hourly_values = getattr_(
+                inlet_temperature_schedule, "schedules", "hourly_values"
+            )
+        if is_heat_recovered_by_drain:
+            drain_heat_recovery_efficiency = getattr_(
+                distribution_system,
+                "service_water_heating_distribution_systems",
+                "drain_heat_recovery_efficiency",
+            )
+
+            assert_(
+                0.0 <= drain_heat_recovery_efficiency <= 1.0,
+                "`drain_heat_recovery_efficiency` value must be between 0 and 1.",
+            )
+
+    space_ids = get_spaces_served_by_swh_use(rmd, swh_use["id"])
+    space_within_building_segment_ids = find_all(
+        "$.zones[*].spaces[*].id", building_segment
+    )
+    spaces = [
+        find_exactly_one_space(rmd, space_id)
+        for space_id in space_ids
+        if space_id in space_within_building_segment_ids
+    ]
+
+    if not spaces and use_units not in REQUIRED_USE_UNIT:
+        spaces = find_all("$.zones[*].spaces[*]", building_segment)
+
+    hourly_values = (
+        getattr_(hourly_schedule, "hourly_schedule", "hourly_values")
+        if hourly_schedule is not None
+        else [1] * len(inlet_temperature_hourly_values)
+    )
+
+    swh_use_value = swh_use.get("use", 0.0)
+    energy_required_by_space = {}
+    volume = ZERO.VOLUME
+    for space in spaces:
+        space_id = space["id"]
+        if use_units == SERVICE_WATER_HEATING_USE_UNIT.POWER_PER_PERSON:
+            power = swh_use_value * ureg("W") * space.get("number_of_occupants", 0)
+            energy_required = (
+                power
+                * sum(hourly_values)
+                * ureg("hr")
+                * (1 - drain_heat_recovery_efficiency)
+            )
+            energy_required_by_space[space_id] = energy_required.to("Btu")
+
+        elif use_units == SERVICE_WATER_HEATING_USE_UNIT.POWER_PER_AREA:
+            power = swh_use_value * ureg("W/m2") * space.get("floor_area", ZERO.AREA)
+            energy_required = (
+                power
+                * sum(hourly_values)
+                * ureg("hr")
+                * (1 - drain_heat_recovery_efficiency)
+            )
+            energy_required_by_space[space_id] = energy_required.to("Btu")
+
+        elif use_units == SERVICE_WATER_HEATING_USE_UNIT.POWER:
+            energy_required_by_space[space_id] = (
+                swh_use_value
+                * ureg("W")
+                * sum(hourly_values)
+                * ureg("hr")
```

Comment:
Maybe add bracket to indicate the unit is per numerical values?
```python
(swh_use_value
                * ureg("W"))
                * (sum(hourly_values)
                * ureg("hr"))
```

Reply:
Good point. Brackets are added here and in Line 199-201